lib/AST/ASTContext.cpp

```diff
@@ -4461,16 +4461,33 @@ Type ASTContext::getBridgedToObjC(const DeclContext *dc, Type type,
   return Type();
 }
 
-const clang::Type *
-ASTContext::getClangFunctionType(ArrayRef<AnyFunctionType::Param> params,
-                                 Type resultTy,
-                                 FunctionTypeRepresentation trueRep) {
+void
+ASTContext::initializeClangTypeConverter() {
   auto &impl = getImpl();
   if (!impl.Converter) {
     auto *cml = getClangModuleLoader();
     impl.Converter.emplace(*this, cml->getClangASTContext(), LangOpts.Target);
   }
-  return impl.Converter.getValue().getFunctionType(params, resultTy, trueRep);
+}
+
+const clang::Type *
+ASTContext::getClangFunctionType(ArrayRef<AnyFunctionType::Param> params,
+                                 Type resultTy,
+                                 FunctionTypeRepresentation trueRep) {
+  initializeClangTypeConverter();
+  return getImpl().Converter.getValue().getFunctionType(params, resultTy,
```

Comment:
This looks like it would be cleaner if you had a getClangTypeConverter() that checked if it needs to be initialized and also returned `Converter`

Reply:
Fixed.